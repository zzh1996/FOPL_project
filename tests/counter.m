Begin 
    Function makecounter Paras init
    Begin
        Function counter Paras inc
        Begin
            Assign init Plus init inc
            Return init
        End
        Return counter
    End

    Var ca cb End
    Assign ca Apply makecounter Argus 1 End
    Print Apply ca Argus 2 End
    Assign cb Apply makecounter Argus 10 End
    Print Apply cb Argus 10 End
    Print Apply ca Argus 5 End
    Print Apply cb Argus 80 End
End
