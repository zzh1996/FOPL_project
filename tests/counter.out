3 20 8 100
